### PR TITLE
Feature/restructure for welsh unavailable reason

### PR DESCRIPTION
**Pull request by philratcliffe**
### What

Create new unavailable placeholder lookups for unavailable reasons for both LEO and other stats.

Restructure unavailable reason handling during stats building for both English and Welsh reason strings.

Update so common code handling string building from data, agg etc by dynamically switching between lookups at run time.

Updated many test fixtures so that tests would pass with new unavailable structure.

### How to review

Review the changes

Run the code and check the structure is as expected with the placeholder -  the placeholder will be replaced when we receive the translated reason codes.

### CODE REVIEW

File: EtlPipelineBlobTrigger/course_stats.py [hunk 1]
```diff
@@ -730,72 +767,70 @@ def need_unavailable(self, xml_elem):
...
+            unavail_reason_code, subj_key, agg, elem
+        )
+        unavailable["reason_welsh"] = self.get_unavailable_reason_str(
+            unavail_reason_code, subj_key, agg, elem, welsh=True
+        )
+        return unavailable
 
     def get_unavailable_reason_str(
-        self, unavail_reason_code, subj_key, agg, xml_elem
+        self, unavail_reason_code, subj_key, agg, xml_elem, welsh=False
```
Comment (nshumoogum) [inline EtlPipelineBlobTrigger/course_stats.py]: If im understanding this correctly, should welsh really be hardcoded to False? Or should the logic actually be add the welsh unavailability reason even if welsh lookup is missing a reason (e.g. empty string)?

File: EtlPipelineBlobTrigger/course_stats.py [hunk 2]
```diff
@@ -730,72 +767,70 @@ def need_unavailable(self, xml_elem):
...
-        ]
+        validate_agg(unavail_reason_code, agg, unavail_reason_lookup)
+        partial_reason_str = unavail_reason_lookup["data"][
+            unavail_reason_code
+        ][agg]
         partial_reason_str = unicodedata.normalize("NFKD", partial_reason_str)
-        subj = self.get_unavailable_reason_subj(subj_key)
+        if welsh:
+            subj = self.get_unavailable_reason_subj_welsh(subj_key)
+            subj = ""  # TODO Remove this line when translations avaialble
```
Comment (nshumoogum) [inline EtlPipelineBlobTrigger/course_stats.py]: Not sure why you would need this, just return empty string from lookup?

Comment (nshumoogum) [reply EtlPipelineBlobTrigger/course_stats.py]: Never mind I see why you've done this now from looking at the welsh lookup table.

File: EtlPipelineBlobTrigger/lookup_files/unavailreason_english.json [hunk 3]
```diff
@@ -30,5 +30,7 @@
             "22": "There is no data available for this course. This is because the course has not yet run or has not been running long enough for this data to be available. For this reason, the data displayed is for students on other courses in ",
             "23": "There is no data available for this course. This is because the course has not yet run or has not been running long enough for this data to be available. For this reason, the data displayed is for students on other courses in "
         }
-    }
+    },
+    "no-subject": "this subject",
+    "agg-over-two-years": " across the last two years."
```
Comment (nshumoogum) [inline EtlPipelineBlobTrigger/lookup_files/unavailreason_english.json]: I like that this has been moved out of the code :)

File: EtlPipelineBlobTrigger/course_stats.py (hunk 1 above)
Comment (philratcliffe) [reply EtlPipelineBlobTrigger/course_stats.py]: This is using Python's default argument values. If the caller doesn't specify a value for welsh it defaults to False so the english lookup will be used.

**Pull request merged**